src/OpenTelemetry.Collector.Dependencies/Implementation/HttpDesktopDiagnosticListener.net461.cs

```diff
@@ -0,0 +1,155 @@
+// </copyright>
+#if NET461
+using System;
+using System.Diagnostics;
+using System.Net;
+using OpenTelemetry.Context.Propagation;
+using OpenTelemetry.Trace;
+
+namespace OpenTelemetry.Collector.Dependencies.Implementation
+{
+    internal class HttpDesktopDiagnosticListener : ListenerHandler
+    {
+        private readonly PropertyFetcher startRequestFetcher = new PropertyFetcher("Request");
+        private readonly PropertyFetcher stopResponseFetcher = new PropertyFetcher("Response");
+        private readonly PropertyFetcher stopExceptionFetcher = new PropertyFetcher("Exception");
+        private readonly HttpClientCollectorOptions options;
+
+        public HttpDesktopDiagnosticListener(Tracer tracer, HttpClientCollectorOptions options)
+            : base(System.Diagnostics.HttpHandlerDiagnosticListener.DiagnosticListenerName, tracer)
+        {
+            this.options = options;
+        }
+
+        public override void OnStartActivity(Activity activity, object payload)
+        {
+            const string EventNameSuffix = ".OnStartActivity";
+            if (!(this.startRequestFetcher.Fetch(payload) is HttpWebRequest request))
+            {
+                CollectorEventSource.Log.NullPayload(nameof(HttpDesktopDiagnosticListener) + EventNameSuffix);
+                return;
+            }
+
+            this.Tracer.StartActiveSpanFromActivity(request.RequestUri.AbsolutePath, activity, SpanKind.Client, out var span);
+
+            if (span.IsRecording)
+            {
+                span.PutComponentAttribute("http");
+                span.PutHttpMethodAttribute(request.Method.ToString());
+                span.PutHttpHostAttribute(request.RequestUri.Host, request.RequestUri.Port);
+                span.PutHttpRawUrlAttribute(request.RequestUri.OriginalString);
+
+                if (this.options.SetHttpFlavor)
+                {
+                    span.PutHttpFlavorAttribute(request.ProtocolVersion.ToString());
+                }
+            }
+
+            if (!(this.options.TextFormat is TraceContextFormat))
+            {
+                this.options.TextFormat.Inject<HttpWebRequest>(span.Context, request, (r, k, v) => r.Headers.Add(k, v));
+            }
+        }
+
+        public override void OnStopActivity(Activity activity, object payload)
+        {
+            const string EventNameSuffix = ".OnStopActivity";
+            var span = this.Tracer.CurrentSpan;
+
+            if (span == null || !span.Context.IsValid)
+            {
+                CollectorEventSource.Log.NullOrBlankSpan(nameof(HttpDesktopDiagnosticListener) + EventNameSuffix);
+                return;
+            }
+
+            if (span.IsRecording)
+            {
+                if (this.stopResponseFetcher.Fetch(payload) is HttpWebResponse response)
+                {
+                    span.PutHttpStatusCode((int)response.StatusCode, response.StatusDescription);
+                }
+            }
+
+            span.End();
```

Comment:
let's `End` the span even if the `Fetch` on statement above failed. Please add `try{}catch`. This is one of those issues which will be very hard to troubleshoot

Reply:
Done